lib/pbio/drv/legodev/legodev_pup.c

```diff
@@ -394,25 +407,28 @@ void pbdrv_legodev_init(void) {
     pbdrv_legodev_pup_uart_process_start();
 }
 
+static pbdrv_legodev_dev_t *pbdrv_legodev_get_parent_of(pbdrv_legodev_pup_uart_dev_t *uartdev) {
+    for (uint8_t i = 0; i < PBIO_ARRAY_SIZE(devs); i++) {
+        pbdrv_legodev_dev_t *dev = &devs[i];
+        if (!dev->is_internal && dev->ext_dev->uart_dev == uartdev) {
+            return dev;
+        }
+    }
+    return NULL;
+}
+
 /**
- * Resets device detection. Called by legodev_uart if a device gets out of sync, usually by unplugging.
+ * Resets device connection watchdog timer. This allows the detection manager
+ * to resume when a UART device stopped working, usually by being unplugged.
  *
  * @param [in]  dev           legodev device.
  */
-void pbdrv_legodev_pup_reset_device_detection(pbdrv_legodev_pup_uart_dev_t *uartdev) {
-    for (uint8_t i = 0; i < PBIO_ARRAY_SIZE(devs); i++) {
-        pbdrv_legodev_dev_t *dev = &devs[i];
-        if (dev->is_internal) {
-            // Internal devices don't have a uart device.
-            continue;
-        }
-        if (dev->ext_dev->uart_dev == uartdev) {
-            // On match reset device connection manager state.
-            dev->ext_dev->connected_type_id = PBDRV_LEGODEV_TYPE_ID_NONE;
-            dev->ext_dev->dcm.dev_id_match_count = 0;
-            return;
-        }
+void pbdrv_legodev_pup_reset_watchdog(pbdrv_legodev_pup_uart_dev_t *uartdev) {
+    pbdrv_legodev_dev_t *dev = pbdrv_legodev_get_parent_of(uartdev);
+    if (dev == NULL || dev->is_internal) {
```

Comment:
Isn't the `is_internal` check redundant?

Is it even possible to have an "internal" uartdev?